.github/actions/compile-commit/action.yml

```diff
@@ -0,0 +1,32 @@
+name: compile-commit
+description: "Compile specific commit"
+inputs:
+  commit:
+    description: "Git commit to compile"
+  base_ref:
+    description: 'Git base ref'
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # checkout all commits to be able to determine merge base
+        ref: ${{ inputs.commit }}
+    - uses: ./.github/actions/setup
+    - name: Check if a specified commit compiles
+      shell: bash
+      run: |
+        set -x
+        # allow overriding Maven command
+        export MAVEN="./mvnw"
+        # maven.wagon.rto is in millis, defaults to 30m
+        export MAVEN_OPTS="-Xmx3G -XX:+ExitOnOutOfMemoryError -Dmaven.wagon.rto=60000"
+        export MAVEN_COMPILE_COMMITS="-B --strict-checksums --quiet -T C1 -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all=true -Dmaven.javadoc.skip=true --no-snapshot-updates --no-transfer-progress -pl !:trino-server-rpm"
+        export MAVEN_GIB="-P gib -Dgib.referenceBranch=refs/remotes/origin/${{ inputs.base_ref }}"
+        $MAVEN package ${MAVEN_COMPILE_COMMITS} ${MAVEN_GIB}
+    - name: Clean local Maven repo
```

Comment:
Thanks for pointing that out. I will remove the outside cleanup since there's no `setup` call in the `check-commits-dispatcher` job. 

I'd rather not inline Maven-related variables since it makes keeping this definition and the `env` block from `ci.yml` in sync easier by just copy-pasting. With inlined vars you'd have to know which option came from which env var and it gets tricky.

Reply:
Maybe there should be some sensible defaults here and the rest would be passed as an input parameter.

Reply:
If the values for these parameters come from a workflow from `ci.yml` then their values will be locked to whatever is on PR HEAD, and we'll face the same problem all over again if they change in the middle of the PR. The fix relies on this composite job being entirely standalone, and checked out from the correct commit before being executed.

Reply:
This is worth putting in a comment.